diagnostic_updater/include/diagnostic_updater/diagnostic_updater.hpp

```diff
@@ -370,11 +370,33 @@ class Updater : public DiagnosticTaskVector
    * \param h Node handle from which to get the diagnostic_period
    * parameter.
    */
-  explicit Updater(rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("diagnostics_updater"))
-  : node_(node), node_name_(node_->get_name())
+  template<class NodeT>
+  explicit Updater(std::shared_ptr<NodeT> node)
+  : Updater(
+      node->get_node_base_interface(),
+      node->get_node_topics_interface(),
+      node->get_node_logging_interface(),
+      node->get_node_parameters_interface())
+  {}
+
+  Updater(
+    std::shared_ptr<rclcpp::node_interfaces::NodeBaseInterface> base_interface,
+    std::shared_ptr<rclcpp::node_interfaces::NodeTopicsInterface> topics_interface,
+    std::shared_ptr<rclcpp::node_interfaces::NodeLoggingInterface> logging_interface,
+    std::shared_ptr<rclcpp::node_interfaces::NodeParametersInterface> parameters_interface)
+  : verbose_(false),
+    publisher_(
+      rclcpp::create_publisher<diagnostic_msgs::msg::DiagnosticArray>(
+        topics_interface, "/diagnostics", 1)),
+    logger_(logging_interface->get_logger()),
+    node_name_(base_interface->get_name()),
+    warn_nohwid_done_(false)
   {
-    // @todo: how to deal with default node?
-    setup();
+    double period = parameters_interface->declare_parameter(
```

Comment:
This doesn't seem right - why is the node name part of the parameter name?

Also, why is the parameter name changing from `diagnostics_update`? A better question is why it isn't the same as it was in ROS 1 (`diagnostic_period`).

Reply:
In addition, this parameter name no longer matches the python code. At the very least, they should be the same...

Reply:
The node name really is meant to be the namespace part of the parameter. I think we were discussing about this offline, but I believe it makes sense to namespace this parameter in case there are multiple diagnostic_updater instances.

Reply:
Looking at the code, I think the parameter in C++ was never en pair with Python. I am happy to address this though.

Reply:
https://github.com/ros/diagnostics/pull/112/commits/4edd6dfe6af8950d63d29d9f436bef87a015c316 should make them equal

Reply:
The whole duplicated node name thing still doesn't sit well with me. Here is an example of the way this looks in one of my use cases:
```
$ ros2 param list
/test_driver:
  angular_velocity_stdev
  calibration.accelerometer_offset
  calibration.accelerometer_radius
  calibration.gyroscope_offset
  calibration.magnetometer_offset
  calibration.magnetometer_radius
  frame_id
  frequency
  linear_acceleration_stdev
  magnetic_field_stdev
  orientation_stdev
  port
  self_manage
  test_driver.diagnostic_period
  use_magnetometer
  use_sim_time
$ ros2 param get /test_driver test_driver.diagnostic_period
Double value is: 1.0
```

The parameter already falls explicitly under the node. What does the duplicated prefix buy us?

Reply:
You're absolutely right. That is nonsense. We had the discussion about namespacing it though to make sure that its original name `period` would indicate that it's a period coming from diagnostics. 
But you're right it shouldn't be the node name, it should actually just be `diagnostic_updater.period`. What do you think?

Reply:
https://github.com/ros/diagnostics/pull/112/commits/691770ca9dc757ef3ad5ded8b41d154530a7c57c 

your example would turn into this now:
```
$ ros2 param list
/test_driver:
  diagnostic_updater.period
```

Which makes a lot of sense to me.

Reply:
Okay, this checks out. Thanks for iterating with me.